### PR TITLE
Permission editor blogpost

### PR DESCRIPTION
**Pull request by enikonemeth**

### CODE REVIEW

File: _posts/2020-09-21-permission-editor.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+
+title: "Permission editor in admin ui"
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
title: "Permission editor on the admin ui"
```

File: _posts/2020-09-21-permission-editor.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+image: "../img/posts/permission_editor.jpg"
+tags: [sensenet, permission]
+
+---
+
+"Nobody can hurt me without my permission. /Mahatma Gandhi/"
+
+---
+
+In the IT world you often handle secure data, which you don't want to share with the whole universe. But how can you hide those from the outsiders? What if you would like to share those only with your group? And another data with another group?
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
In the IT world you often handle sensitive data, which you don't want to share with the whole universe. But how you can hide those contents from outsiders? What if you would like to share those things only with a specific group of users? Or other data with another group? 
```

File: _posts/2020-09-21-permission-editor.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+---
+
+"Nobody can hurt me without my permission. /Mahatma Gandhi/"
+
+---
+
+In the IT world you often handle secure data, which you don't want to share with the whole universe. But how can you hide those from the outsiders? What if you would like to share those only with your group? And another data with another group? 
+
+Of course there is a simple solution for that: permission system. Almost every software has permissions which handle what u can see & do.
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
Of course there is a common solution for that: permission system. Almost every software has permissions which handle what users can see or do.
```

File: _posts/2020-09-21-permission-editor.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+I think most of us use any social network. You may search for a person and you cannot see her/his photos, after you apply a friend request everything change.
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
I think most of u use some kind of social media. Think about searching for someone and that by default you cannot see her/his photos, but after you apply a friend request everything changes.
```

File: _posts/2020-09-21-permission-editor.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+You see more than ever. Or what about your company? If you are not the boss I am sure that your company has datas which you do not have the sufficient permission.
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
And what about your company data? Unless you are the boss, I am sure that your company has data for which you do not have sufficient permissions.
```

File: _posts/2020-09-21-permission-editor.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+Permissions are everywhere. And of course sensenet has its own permission system as well. On the new admin-ui gui we did not have any surface to handle these permissions. Till now. 🕛 In the last couple of weeks we have tried to design and implement the simplest, easiest to use **permission editor**.
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
Permissions are everywhere. And of course sensenet has its own permission system as well. On the new admin surface 
handling permissions was only possible with coding and running commands. Till now. 🕛 In the last couple of weeks we were about to design and implement the simplest, easiest to use **permission editor**.
```

File: _posts/2020-09-21-permission-editor.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+On the permission editor dialog you can see the groups and permissions based on these settings. 
+
+<p align="center">
+<img src="/img/posts/permission_groups.png">
+</p>
+
+## How to use it? 🎮
+
+Select a content from your repository, right-click on it and select 'Set permissions' from the dropdown. On the permission view you can check the entries which has inherited or set permissions. Selecting an entry makes the permission editor dialog open.
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
Select a content from your repository, right-click on it and select 'Set permissions' from the dropdown. On the permission view you can check the permission entries (users or groups) which are inherited or set directly on the current content. Selecting an entry makes the permission editor dialog open.
```

File: _posts/2020-09-21-permission-editor.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+On the dialog turn the switchers to on/off to enable/disable a permission.  On groups there are also switchers as well: you can easily enable/disable all of the permission under the selected group... and there is the king of kings: Full access 👑 (you can turn on/off all permissions)
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
On the dialog use the switchers to enable/disable a permission. There are also group level switchers, so you can easily enable/disable all the permissions in the selected group...and there is the king of kings: Full access 👑 with which you can turn all permissions on/off.
```

File: _posts/2020-09-21-permission-editor.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+Some permissions are disabled because they are inherited and you cannot override them. The changes only apply on clicking Submit button so for your comfort there is also a reset button to restore the unsaved changes.
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
When you open the dialog some permissions are disabled because they are inherited from a parent and you cannot override them. 
Changes only apply on clicking Submit button so for your comfort there is also a reset button to restore the unsaved changes.
```

File: _posts/2020-09-21-permission-editor.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+<p align="center">
+<img src="/img/posts/permission_how_to_use.gif">
+</p>
+
+## Additional features 💄👓👜
+
+**Make content public/private**
+
+If you would like to share your content (content can be almost anything, cause in sensenet everything is a content) with somebody e.g. from another company who has no authority to your repository you can do it easily by clicking on "Make content public" button. If you change your mind reverting this is easy as it was to set.
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
If you would like to share your content (content can be almost anything, cause in sensenet everything is a content) with visitors who are not users in your repository you can do it easily by clicking on "Make content public" button. If you change your mind reverting this is easy as it was to set. This thing come in handy when you build a public website and most of your repository data is shared with all the visitors.
```

File: _posts/2020-09-21-permission-editor.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+
+<p align="center">
+<img src="/img/posts/permission_make_public.gif">
+</p>
+
+**Assign new permission**
+
+If you would like to add permission for someone who has no permisison at all on this content you can do it with assign new permission button. Select the user who needs the rights and turn on/off the switchers 😉
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
If you would like to grant permission for someone on a content who has no permission setting at all, you can do it using the _Assign new permission_ button. Search for and select the user or group to whom you want to set permissions and turn on/off the switchers. 😉
```

File: _posts/2020-09-21-permission-editor.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+
+<p align="center">
+<img src="/img/posts/permission_assign_new.gif">
+</p>
+
+**Local only**
+
+There are cases however when you do not want child content to inherit a permission entry. For example you want to allow certain users to see a content (e.g. a Content List) but you do not want them to be able to see content that were added to that list. You can mark the permission entry as 'local-only' to avoid permission inheritence.
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
One of the big advantages of sensenet content repository tree is inheritance that is applied on permissions as well. However, there are cases when you do not want child contents to inherit permissions. For example you want to allow certain users to see a content (e.g. a Content List) but you do not want them to be able to see contents that are inside this container. To avoid permission inheritance, mark permission entries as 'local-only'.
```

File: _posts/2020-09-21-permission-editor.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+On permission view we mark these local-only entries with an icon.
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
On the permission view these local-only entries are marked with an icon.
```

File: _posts/2020-09-21-permission-editor.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+Some of the built-in permissions requires to allow other permission(s). E.g: If you would like to set Preview permission to an entry it requires to have See permission as well. You don't need to handle these relations cause it is automatic in the background. Setting the Preview permission will set See permission as well.
+
+<p align="center">
+<img src="/img/posts/permission_force_relations.gif">
+</p>
+
+
+## Must read 📚🔖
+
+More details about our permission system:
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
More details about sensenet's permission system:
```

File: _posts/2020-09-21-permission-editor.md [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+Of course there is a simple solution for that: permission system. Almost every software has permissions which handle what u can see & do.
+
+I think most of us use any social network. You may search for a person and you cannot see her/his photos, after you apply a friend request everything change.
+You see more than ever. Or what about your company? If you are not the boss I am sure that your company has datas which you do not have the sufficient permission.
+
+Permissions are everywhere. And of course sensenet has its own permission system as well. On the new admin-ui gui we did not have any surface to handle these permissions. Till now. 🕛 In the last couple of weeks we have tried to design and implement the simplest, easiest to use **permission editor**.
+
+## Customize it as you want 🔧🔨
+
+sensenet has a lot of built-in permission and we also keep some for your special needs:  Custom01, Custom02, etc. You can also group these permissions for the permission editor dialog as you want. The related settings file is available under Setup menu / Permission.settings.
```
Comment (herflis) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
sensenet has a lot of built-in permissions but there are some reserved one that you are free to use in case of you have special needs ([custom permissions](https://docs.sensenet.com/concepts/user-and-permission-management/04-custom-roles-and-permissions#permissions)). You can also group these permissions for the permission editor dialog as you want. The related settings file is available under Setup menu / Permission.settings.
```

**Review by herflis — CHANGES_REQUESTED** [review]

File: _posts/2020-09-21-permission-editor.md [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+
+I think most of u use some kind of social media. Think about searching for someone and that by default you cannot see her/his photos, but after you apply a friend request everything changes.
+And what about your company data? Unless you are the boss, I am sure that your company has data for which you do not have sufficient permissions.
+
+Permissions are everywhere. And of course sensenet has its own permission system as well. On the new admin surface 
+handling permissions was only possible with coding and running commands. Till now. 🕛 In the last couple of weeks we were about to design and implement the simplest, easiest to use **permission editor**.
+
+## Customize it as you want 🔧🔨
+
+sensenet has a lot of built-in permissions but there are some reserved one that you are free to use in case of you have special needs ([custom permissions](https://docs.sensenet.com/concepts/user-and-permission-management/04-custom-roles-and-permissions#permissions)). You can also group these permissions for the permission editor dialog as you want. The related settings file is available under Setup menu / Permission.settings.
```
Comment (molnarm84) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
sensenet has a lot of built-in permissions but there are some reserved ones that you are free to use in case of you have special needs ([custom permissions](https://docs.sensenet.com/concepts/user-and-permission-management/04-custom-roles-and-permissions#permissions)). You can also group these permissions for the permission editor dialog as you want. The related settings file is available under Setup menu / Permission.settings.
```

File: _posts/2020-09-21-permission-editor.md [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+
+"Nobody can hurt me without my permission. /Mahatma Gandhi/"
+
+---
+
+In the IT world you often handle sensitive data, which you don't want to share with the whole universe. But how you can hide those contents from outsiders? What if you would like to share those things only with a specific group of users? Or other data with another group? 
+
+Of course there is a common solution for that: permission system. Almost every software has permissions which handle what users can see or do.
+
+I think most of u use some kind of social media. Think about searching for someone and that by default you cannot see her/his photos, but after you apply a friend request everything changes.
```
Comment (huszerlbence) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
I think most of us use some kind of social media. Think about searching for someone and that by default you cannot see her/his photos, but after you apply a friend request everything changes.
```

File: _posts/2020-09-21-permission-editor.md [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+And what about your company data? Unless you are the boss, I am sure that your company has data for which you do not have sufficient permissions.
+
+Permissions are everywhere. And of course sensenet has its own permission system as well. On the new admin surface 
+handling permissions was only possible with coding and running commands. Till now. 🕛 In the last couple of weeks we were about to design and implement the simplest, easiest to use **permission editor**.
```
Comment (huszerlbence) [inline _posts/2020-09-21-permission-editor.md:22]: "On the new admin surface handling permissions was only possible with coding and running commands."

I'm not sure about this but the coding and command running didn't happened on the admin ui itself. If that's correct I would rephrase it a bit.

File: _posts/2020-09-21-permission-editor.md [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+
+## Customize it as you want 🔧🔨
+
+sensenet has a lot of built-in permissions but there are some reserved ones that you are free to use in case of you have special needs ([custom permissions](https://docs.sensenet.com/concepts/user-and-permission-management/04-custom-roles-and-permissions#permissions)). You can also group these permissions for the permission editor dialog as you want. The related settings file is available under Setup menu / Permission.settings.
```
Comment (huszerlbence) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
sensenet has a lot of built-in permissions but there are some reserved ones that you are free to use in case you have special needs ([custom permissions](https://docs.sensenet.com/concepts/user-and-permission-management/04-custom-roles-and-permissions#permissions)). You can also group these permissions for the permission editor dialog as you want. The related settings file is available under Setup menu / Permission.settings.
```

Comment (huszerlbence) [inline _posts/2020-09-21-permission-editor.md]: "You can also group these permissions for the permission editor dialog as you want"

I don't understand this part, please explain (:

File: _posts/2020-09-21-permission-editor.md [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+
+On the permission editor dialog you can see the groups and permissions based on these settings. 
+
+<p align="center">
+<img src="/img/posts/permission_groups.png">
+</p>
+
+## How to use it? 🎮
+
+Select a content from your repository, right-click on it and select 'Set permissions' from the dropdown. On the permission view you can check the permission entries (users or groups) which are inherited or set directly on the current content. Selecting an entry makes the permission editor dialog open.
```
Comment (huszerlbence) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
Select a content from your repository, right-click on it and select 'Set permissions' from the dropdown. On the permission view you can check the permission entries (users or groups) which are inherited or set directly on the current content. Selecting an entry opens the permission editor dialog.
```

File: _posts/2020-09-21-permission-editor.md [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+
+On the dialog use the switchers to enable/disable a permission. There are also group level switchers, so you can easily enable/disable all the permissions in the selected group...and there is the king of kings: Full access 👑 with which you can turn all permissions on/off.
```
Comment (huszerlbence) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
On the dialog use the switches to enable/disable a permission. There are also group level switches, so you can easily enable/disable all the permissions in the selected group...and there is the king of kings: Full access 👑 with which you can turn all permissions on/off.
```

File: _posts/2020-09-21-permission-editor.md [hunk 22]
```diff
@@ -0,0 +1,111 @@
...
+
+If you would like to share your content (content can be almost anything, cause in sensenet everything is a content) with somebody e.g. from another company who has no authority to your repository you can do it easily by clicking on "Make content public" button. If you change your mind reverting this is easy as it was to set.
+
+<p align="center">
+<img src="/img/posts/permission_make_public.gif">
+</p>
+
+**Assign new permission**
+
+If you would like to grant permission for someone on a content who has no permission setting at all, you can do it using the _Assign new permission_ button. Search for and select the user or group to whom you want to set permissions and turn on/off the switchers. 😉
```
Comment (huszerlbence) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
If you would like to grant permission for someone on a content who has no permission setting at all, you can do it using the _Assign new permission_ button. Search for and select the user or group to whom you want to set permissions and turn on/off the switches. 😉
```

File: _posts/2020-09-21-permission-editor.md [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+
+<p align="center">
+<img src="/img/posts/permission_assign_new.gif">
+</p>
+
+**Local only**
+
+One of the big advantages of sensenet content repository tree is inheritance that is applied on permissions as well. However, there are cases when you do not want child contents to inherit permissions. For example you want to allow certain users to see a content (e.g. a Content List) but you do not want them to be able to see contents that are inside this container. To avoid permission inheritance, mark permission entries as 'local-only'.
```
Comment (huszerlbence) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
One of the big advantages of sensenet content repository tree is inheritance which is applied on permissions as well. However, there are cases when you do not want child contents to inherit permissions. For example you want to allow certain users to see a content (e.g. a Content List) but you do not want them to be able to see contents that are inside this container. To avoid permission inheritance, mark permission entries as 'local-only'.
```

File: _posts/2020-09-21-permission-editor.md [hunk 24]
```diff
@@ -0,0 +1,111 @@
...
+Some of the built-in permissions requires to allow other permission(s). E.g: If you would like to set Preview permission to an entry it requires to have See permission as well. You don't need to handle these relations cause it is automatic in the background. Setting the Preview permission will set See permission as well.
+
+<p align="center">
+<img src="/img/posts/permission_force_relations.gif">
+</p>
+
+
+## Must read 📚🔖
+
+More details about sensenet's permission system:
```
Comment (huszerlbence) [inline _posts/2020-09-21-permission-editor.md]:
```suggestion
More details about sensenet permission system:
```

**Review by huszerlbence — CHANGES_REQUESTED** [review]
I've changed small stuff here and there. And I did not understand a phrase (commented). 
thanks for the opportunity.

**Review by huszerlbence — APPROVED** [review]

**Review by molnarm84 — APPROVED** [review]

**Review by herflis — APPROVED** [review]
